### ISSUE TITLE
multple VerticaBatches don't work

### ISSUE THREAD

**Issue by egorps** [spilgames/pyvertica#15]
First off, let me tell you how much I love this library - it's great to be able to use batches for vertica in python!

I think I found a bug in VerticaBatch - basically, I am trying to open two VerticaBatches simultaneously. Each batch will insert to a different table. I've found that when the _start_batch method is called on the second batch, the program locks up. 

I will dig into the code to see if I can figure out what is going on - if you have any insights, please let me know.

**Comment by egorps** [comment]
A simple test program that demonstrates the problem:

https://gist.github.com/egorps/7677286

**Comment by egorps** [comment]
I did some testing with multiple FIFOs in python by starting up new threads (or processes) and having each read from one FIFO. This test seemed to work, indicating that it should be possible to have two vertica batches going in a single process:

https://gist.github.com/egorps/7695667

I did notice that the call to open the writing end of the fifo will block until a reader is listening. This seems to be the same behavior I am seeing when starting up a second VerticaBatch. I'm wondering if there is a limitation in the vertica driver? Possibly COPY LOCAL commands can only execute one at a time. I will test that out next.

**Comment by lomignet** [comment]
Hi,

Thanks a lot for the feedback! I am very happy to see that pyvertica is used beyond us, and very glad that you take the time to patch and comment.

Regarding the fifo locking, I believe that it is the way fifo works. If you open 2 terminals, on the first one you create and write to the fifo:
mkfifo bar
echo 'foo' > bar
This will block up to you read from the 2nd terminal:
cat bar

On top of my head I do not find a good reason why it should not be possible to get 2 batches in parallel, but I know that this fifo blocking was a bit tricky to get around. I will look more into it on Monday as well. 

There might be one thing to check first: do you have Threading = 1 in your odbc.ini file, section [ODBC] or in your odbcinst.ini in the relevant [VerticaDSN] section? Without this odbc will only accept one query per connection.

I am curious to see how we can fix this,
Cheers

**Comment by egorps** [comment]
That did the trick!

I set up a driver in odbcinst.ini like this:
[Vertica]
Description     = vertica driver
Driver          = /opt/vertica/lib64/libverticaodbc.so
Threading       = 1

And then configured my odbc.ini file like this:

[ceilometer]
Description             = ceilometer database connection
Driver          = Vertica
Database                = som
Servername      = <servername>
UID             = <username>
PWD             = <password>
Port            = 5433
ConnSettings            =
SSLMode         = require

I wrote a small program that breaks the problem down into two individual copy statements. This works with Threading = 1 - I can log into my vertica cluster and see that both queries are running, and the main thread is not blocked waiting.  

Thanks a lot for the suggestion, I'll close this issue.
-john
